app/code/Magebit/Faq/Model/QuestionRepository.php

```diff
@@ -0,0 +1,214 @@
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param HydratorInterface|null $hydrator
+     *
+     * @SuppressWarnings(PHPMD.ExcessiveParameterList)
+     */
+    public function __construct(
+        ResourceQuestion $resource,
+        QuestionFactory $questionFactory,
+        \Magebit\Faq\Api\Data\QuestionInterfaceFactory $dataQuestionFactory,
+        QuestionCollectionFactory $questionCollectionFactory,
+        Data\QuestionSearchResultsInterfaceFactory $searchResultsFactory,
+        DataObjectHelper $dataObjectHelper,
+        DataObjectProcessor $dataObjectProcessor,
+        StoreManagerInterface $storeManager,
+        CollectionProcessorInterface $collectionProcessor,
+        ?HydratorInterface $hydrator = null
+    ) {
+        $this->resource = $resource;
+        $this->questionFactory = $questionFactory;
+        $this->questionCollectionFactory = $questionCollectionFactory;
+        $this->searchResultsFactory = $searchResultsFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->dataQuestionFactory = $dataQuestionFactory;
+        $this->dataObjectProcessor = $dataObjectProcessor;
+        $this->storeManager = $storeManager;
+        $this->collectionProcessor = $collectionProcessor;
+        $this->hydrator = $hydrator ?? ObjectManager::getInstance()->get(HydratorInterface::class);
+    }
+
+    /**
+     * Get question
+     *
+     * @param int $questionId
+     * @return void
+     */
+    public function get($questionId): QuestionInterface
+    {
+        $question = $this->questionFactory->create();
+        $question->load($questionId);
+        if (!$question->getId()) {
+            throw NoSuchEntityException::singleField('id', $questionId);
+        }
+        return $question;
+    }
+
+    /**
+     * Save question data
+     *
+     * @param \Magebit\Faq\Api\Data\QuestionInterface $question
+     * @return Question
+     * @throws CouldNotSaveException
+     */
+    public function save(Data\QuestionInterface $question): Question
+    {
+        if (empty($question->getStoreId())) {
+            $question->setStoreId($this->storeManager->getStore()->getId());
+        }
+
+        if ($question->getId() && $question instanceof Question && !$question->getOrigData()) {
+            $question = $this->hydrator->hydrate(
+                $this->getById($question->getId()),
+                $this->hydrator->extract($question)
+            );
+        }
+
+        try {
+            $this->resource->save($question);
+        } catch (\Exception $exception) {
+            throw new CouldNotSaveException(__($exception->getMessage()));
+        }
+        return $question;
+    }
+
+    /**
+     * Load question data by given question Identity
+     *
+     * @param string $questionId
+     * @return Question
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getById($questionId)
```

Comment:
Repository should contain such methods:

- get
- save
- getList
- delete
- deleteById